### PR TITLE
Update design section

### PR DESCRIPTION
**Pull request by Imranr2**
Tackles part of https://github.com/AY2122S1-CS2103T-W16-2/tp/issues/206

View changes here https://imranr2.github.io/tp/DeveloperGuide.html

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -69,75 +70,84 @@ The sections below give more details of each component.
...
-
-<div markdown="span" class="alert alert-info">:information_source: **Note:** An alternative (arguably, a more OOP) model is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique tag, instead of each `Person` needing their own `Tag` objects.<br>
-
-<img src="images/BetterModelClassDiagram.png" width="450" />
-
-</div>
-
+* stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a
+  `ReadOnlyUserPref` objects.
+* stores the address book data in the following components:
```
Comment (zognin) [inline docs/DeveloperGuide.md:136]:
```suggestion
* stores the application data in the following components:
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -69,75 +70,84 @@ The sections below give more details of each component.
...
+  * a `UniqueClientList` containing `Client`s.
```
Comment (zognin) [inline docs/DeveloperGuide.md:137]: Can we change this for all 4 points?
```suggestion
  * a `UniqueClientList` containing `Client` objects.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -69,75 +70,84 @@ The sections below give more details of each component.
...
+  * a `UniqueIngredientList` containing `Ingredient`s.
+  * a `UniqueOrderList` containing `Order`s.
+  * a `UniqueOrderList` containing `Recipe`s.
```
Comment (zognin) [inline docs/DeveloperGuide.md:140]:
```suggestion
  * a `UniqueRecipeList` containing `Recipe` objects.
```

**Review by zognin — COMMENTED** [review]
Minor nits, along with other diagram changes

**Comment by Imranr2** [comment]
You can view the changes using the link

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -133,11 +134,11 @@ The `Model` component,
...
-* stores the address book data in the following components:
-  * a `UniqueClientList` containing `Client`s.
-  * a `UniqueIngredientList` containing `Ingredient`s.
-  * a `UniqueOrderList` containing `Order`s.
-  * a `UniqueOrderList` containing `Recipe`s.
+* stores the application in the following components:
+  * a `UniqueClientList` containing `Client` objects.
+  * a `UniqueIngredientList` containing `Ingredient` objects.
+  * a `UniqueOrderList` containing `Order` objects.
+  * a `UniqueOrderList` containing `Recipe` objects.
```
Comment (zognin) [inline docs/DeveloperGuide.md:141]:
```suggestion
  * a `UniqueRecipeList` containing `Recipe` objects.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -133,11 +134,11 @@ The `Model` component,
 
 * stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a
   `ReadOnlyUserPref` objects.
-* stores the address book data in the following components:
-  * a `UniqueClientList` containing `Client`s.
-  * a `UniqueIngredientList` containing `Ingredient`s.
-  * a `UniqueOrderList` containing `Order`s.
-  * a `UniqueOrderList` containing `Recipe`s.
+* stores the application in the following components:
```
Comment (zognin) [inline docs/DeveloperGuide.md:137]:
```suggestion
* stores the application data in the following components:
```

**Review by zognin — COMMENTED** [review]
Minor nits

**Review by zognin — APPROVED** [review]
LGTM thanks

**Pull request merged**